ts/Stock/Indicators/BB/BBIndicator.ts

```diff
@@ -0,0 +1,304 @@
+        arrLen = arr.length,
+        std = 0,
+        i = 0,
+        value: number;
+
+    for (; i < arrLen; i++) {
+        value = ((isOHLC ? arr[i][index] : arr[i]) as any) - mean;
+        variance += value * value;
+    }
+    variance = variance / (arrLen - 1);
+
+    std = Math.sqrt(variance);
+    return std;
+}
+
+/* eslint-enable valid-jsdoc */
+
+/**
+ * Bollinger Bands series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.bb
+ *
+ * @augments Highcharts.Series
+ */
+class BBIndicator extends SMAIndicator implements Highcharts.MultipleLinesIndicator {
+    /**
+     * Bollinger bands (BB). This series requires the `linkedTo` option to be
+     * set and should be loaded after the `stock/indicators/indicators.js` file.
+     *
+     * @sample stock/indicators/bollinger-bands
+     *         Bollinger bands
+     *
+     * @extends      plotOptions.sma
+     * @since        6.0.0
+     * @product      highstock
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/bollinger-bands
+     * @optionparent plotOptions.bb
+     */
+    public static defaultOptions: BBOptions = merge(SMAIndicator.defaultOptions, {
+        params: {
+            period: 20,
+            /**
+             * Standard deviation for top and bottom bands.
+             */
+            standardDeviation: 2,
+            index: 3
+        },
+        /**
+         * Bottom line options.
+         */
+        bottomLine: {
+            /**
+             * Styles for a bottom line.
+             */
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line. If not set, it's inherited from
+                 * [plotOptions.bb.color](#plotOptions.bb.color).
+                 *
+                 * @type  {Highcharts.ColorString}
+                 */
+                lineColor: void 0
+            }
+        },
+        /**
+         * Top line options.
+         *
+         * @extends plotOptions.bb.bottomLine
+         */
+        topLine: {
+            styles: {
+                lineWidth: 1,
+                /**
+                 * @type {Highcharts.ColorString}
+                 */
+                lineColor: void 0
+            }
+        },
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span><b> {series.name}</b><br/>Top: {point.top}<br/>Middle: {point.middle}<br/>Bottom: {point.bottom}<br/>'
+        },
+        marker: {
+            enabled: false
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    } as BBOptions)
+
+    /* *
+    *
+    *  Prototype Properties
+    *
+    * */
+
+    public data: Array<BBPoint> = void 0 as any;
+
+    public options: BBOptions = void 0 as any;
+
+    public points: Array<BBPoint> = void 0 as any;
+
+    public init(this: BBIndicator): void {
```

Comment:
The generic comment is incorrect. You have two options to solve this. Remove this generic comment, or replace it with "Properties" and add "Functions". In a class pattern property assignments go on the instance and only functions go on the prototype.